funnel/models/space.py

```diff
@@ -110,23 +120,29 @@ def proposals_all(self):
 
     @property
     def proposals_by_status(self):
-        from .proposal import Proposal, PROPOSALSTATUS
+        from .proposal import Proposal
         if self.subspaces:
             basequery = Proposal.query.filter(Proposal.proposal_space_id.in_([self.id] + [s.id for s in self.subspaces]))
         else:
             basequery = Proposal.query.filter_by(proposal_space=self)
-        return dict((status, basequery.filter_by(status=status).order_by(db.desc('created_at')).all()) for (status, title) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT)
+        all_proposals = basequery.filter(Proposal.state.NOT_DRAFT).order_by(db.desc('created_at')).all()
+        all_by_status = defaultdict(list)
+        for p in all_proposals:
+            # returning title instead of value because this is only used in space.html.jnja2,
+            # and that template requires the status title, not the value
+            all_by_status[p.state.label.title].append(p)
+        return all_by_status
```

Comment:
Titles should not be used as dictionary keys. Don't forget that they are localized strings and could cause you grief in a different language.

Reply:
This will mean importing `PROPOSALSTATUS` into the template then.

Reply:
That only means something else is being done wrong. 😃

Reply:
Can `Proposal.state` itself expose a method for the lookup?

Reply:
As the dict has already been provided to it in definition.

Reply:
Let me think about this. You don't need a lookup as much as you need grouping by state.

Reply:
There's now `Proposal.state.group(items)`.